### ISSUE TITLE
Some sourcemap notes

### ISSUE THREAD

**Issue by bep** [evanw/esbuild#4015]
I revised extended the esbuild integration in [hugo](https://github.com/gohugoio/hugo/releases/tag/v0.140.0) with code splitting (yes, I know about the ordering issue(s)), and had some issues getting source maps working correctly across platform.

**Note that using the Go API it was fairly simple to rewrite the source maps `sources` field, so this isn't a blocker for me, but I thought I make an issue jotting down my experiences.**

Two issues:

1. The only way I found to get the `sources` filenames correct was to use the `outDir` setting and then do `filepath.Join(outdir, pathFromSources)`.
2. Filenames on form `/foo/bar.js` or `c:\\foo\\bar.js` in `sources` works great in Chrome on MacOs, but fails on Windows. Converting it into a URL (`file:`) makes it work for both.

**Comment by evanw** [comment]
Yes, there are several open problems with source maps. Here are some similar ones to what you are bringing up:

* https://github.com/evanw/esbuild/issues/1745
* https://github.com/evanw/esbuild/issues/3183
* https://github.com/evanw/esbuild/issues/3613
* https://github.com/evanw/esbuild/issues/3982

The underlying root cause is esbuild treating these strings as file system paths instead of as URLs. However, changing this will break tools that operate on esbuild's output and that expect file system paths, so I'll need to fix these in a breaking change release.
